### DISCUSSION TITLE
Support kvar unit

### DISCUSSION THREAD

**Discussion by toreamun** [home-assistant/architecture#729] [Entity Models] [answered]
Support for unit **var** (POWER_VOLT_AMPERE_REACTIVE) has just been added by #688 and https://github.com/home-assistant/core/pull/62010. Unit **kvar** (kilo var) is also common as a native unit from electrical meters, and should be added as a supported unit. This is the pattern used by power (watt) where both POWER_WATT and POWER_KILO_WATT is supported.

Meters following national standards of Norway and Sweden uses kvar as a native unit for several reported measurements. This is also the unit used by the OBIS standard (see IEC 62056-61 and https://www.promotic.eu/en/pmdoc/Subsystems/Comm/PmDrivers/IEC62056_OBIS.htm).

**Comment by canton7** [comment]
This has been sat quiet for a while -- would it be possible to get an update?

I'm very interested in support for both kvar and kVA. My integration reports real power in kW, so it would be good to report apparent and reactive power in the same order of units.

**Comment by balazs92117** [comment]
I have a sensor (dsmr - p1 port) that provides data with unit of `kvar`. It would be nice if I could manage it in HA. So I support this proposal, and hope somebody can approve it.

**Comment by MartinHjelmare** [comment]
Please list the integrations that can leverage the kvar unit together with the reactive power sensor device class.

**Reply by canton7** [reply]
foxess_modbus for one, and I suspect other integrations which communicate with solar inverters.

**Reply by farmio** [reply]
Fronius could make use of it
https://github.com/home-assistant/core/blob/cb4917f8805efd9c75f245dc22e29b240d8ba559/homeassistant/components/fronius/sensor.py#L357

**Reply by MartinHjelmare** [reply]
Regarding Fronius, `varh` is a unit for reactive energy, right? Here we're talking about reactive power sensors.

**Reply by farmio** [reply]
Oh sorry 😬 Fronius uses that too, fixed the line in the link above. It's `var` but if user wants to convert it, kvar would be nice.

**Reply by balazs92117** [reply]
https://www.home-assistant.io/integrations/dsmr/ could use it too.

**Comment by mkmer** [comment]
https://www.home-assistant.io/integrations/shelly could use this in the Shelly EM device.

**Comment by skrynklarn** [comment]
Can we please get support for `kVAR`. I don't seems why it should not be supported. It is kilo-VAR, there is other unit that have support for kilo and other like mega etc.
Even esphome supports it https://github.com/esphome/esphome/blob/23dec912ad9aaa651e7d261343d4cfef8e320064/esphome/const.py#L1077

There is a lot of devices that over MQTT or by esphome reads electric meters in Sweden, Norway, Netherlands and maybe more, so don't only look at integrations in Home assistans since that gives not the full picture.

**Comment by MartinHjelmare** [comment, marked as answer]
It's ok to add support for kvar. We welcome the needed PRs in core, dev docs and users docs.

**Comment by skrynklarn** [comment]
PR created https://github.com/home-assistant/core/pull/142558

**Reply by MartinHjelmare** [reply]
Thanks! The PRs have been merged.
